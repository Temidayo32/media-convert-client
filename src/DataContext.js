import React, { createContext, useState, useContext } from 'react';

const DataContext = createContext();

export const DataProvider = ({ children }) => {
  const [showSignUpOptions, setShowSignUpOptions] = useState(false);
  const [uploadedVideos, setUploadedVideos] = useState([]);
  const [uploadedImages, setUploadedImages] = useState([]);
  const [downloadPageActive, setDownloadPageActive] = useState(false);
  const [idToken, setIdToken] = useState(null);
  const [emailVerified, setEmailVerified] = useState(false); 
  const [userCredentials, setUserCredentials] = useState(null);
  const [displayType, setDisplayType] = useState('');
  const [oversizedFiles, setOversizedFiles] = useState([]);
  const [showErrorMessages, setShowErrorMessages] = useState(false);
  const [showUploadForm, setShowUploadForm] = useState(true);
  

  return (
    <DataContext.Provider value={{ showSignUpOptions, setShowSignUpOptions, uploadedVideos, setUploadedVideos, uploadedImages, setUploadedImages, downloadPageActive, displayType, setDisplayType, setDownloadPageActive, idToken, setIdToken, emailVerified, setEmailVerified, userCredentials, setUserCredentials, oversizedFiles, setOversizedFiles, showErrorMessages, setShowErrorMessages, showUploadForm, setShowUploadForm }}>
      {children}
    </DataContext.Provider>
  );
};

export const useData = () => useContext(DataContext);
