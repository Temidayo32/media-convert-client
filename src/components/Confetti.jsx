import React, { useEffect } from 'react';
import Particles, { initParticlesEngine } from '@tsparticles/react';
// import { loadSlim } from "@tsparticles/slim";
import { loadFull } from "tsparticles";

function Confetti() {
    useEffect(() => {
        const initializeParticles = async () => {
          await initParticlesEngine(async (engine) => {
            await loadFull(engine);
          });
        };
    
        initializeParticles();
      }, []);

  const options = {
    background: {
      color: {
        value: "transparent",
      },
    },
    fullScreen: {
        enable: true,
        zIndex: 1,
    },
    emitters: {
        position: {
          x: 50,
          y: 100
        },
        rate: {
          quantity: 5,
          delay: 0.15
        }
      },
    particles: {
      number: {
        value: 100,
      },
      color: {
        value: ["#4285F4", "#DB4437", "#0F9D58", "#5C6BC0"],
      },
      shape: {
        type: ['circle', 'square', 'triangle'],
      },
      size: {
        value: 6,
        random: true,
      },
      move: {
        decay: 0.05,
        direction: "top",
        enable: true,
        gravity: {
          enable: true
        },
        outModes: {
          top: "none",
          default: "destroy"
        },
        speed: {
          min: 50,
          max: 100
        }
      },
      opacity: {
        value: 1
      },
      rotate: {
        value: {
          min: 0,
          max: 360
        },
        direction: "random",
        animation: {
          enable: true,
          speed: 30
        }
      },
      tilt: {
        direction: "random",
        enable: true,
        value: {
          min: 0,
          max: 360
        },
        animation: {
          enable: true,
          speed: 30
        }
      },
      size: {
        value: 3,
        animation: {
          enable: true,
          startValue: "min",
          count: 1,
          speed: 16,
          sync: true
        }
      },
      roll: {
        darken: {
          enable: true,
          value: 25
        },
        enlighten: {
          enable: true,
          value: 25
        },
        enable: true,
        speed: {
          min: 5,
          max: 15
        }
      },
      wobble: {
        distance: 30,
        enable: true,
        speed: {
          min: -7,
          max: 7
        }
      },
      responsive: [
        {
          maxWidth: 1024,
          options: {
            particles: {
              move: {
                speed: {
                  min: 33,
                  max: 66
                }
              }
            }
          }
        }
      ]
    },
    preset: "confetti",
  };

  const particlesLoaded = (container) => {
    console.log(container);
  };

  return (
    <div className="absolute top-0 left-0 w-full h-full z-10">
        <Particles
            id="tsparticles"
            options={options}
            particlesLoaded={particlesLoaded}
    />
    </div>
  );
}

export default Confetti;
