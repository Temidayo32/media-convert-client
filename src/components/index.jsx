export {default as AccountSuccess} from './home/AccountSuccess';
export {default as Card } from './home/Card';
export {default as Header} from './home/Header';
export { default as HeaderOptions} from './home/HeaderOptions';
export {default as HeroSection} from './home/HeroSection';
export {default as Login} from './home/Login';
export {default as LoginWithEmail } from './home/LoginWithEmail';
export {default as Signup} from './home/Signup';
export {default as SignUpWithEmail} from './home/SignUpWithEmail';
export {default as SignUpWithGoogle } from './home/SignUpWithGoogle';
export {default as PasswordReset} from './home/PasswordReset';
export {default as EmailVerification} from "./home/EmailVerification";
export {default as EmailVerificationRedirect} from './home/EmailVerificationRedirect';
export { default as Menu } from './home/Menu';
export {default as EmailVerificationMenu} from './home/EmailVerificationMenu';
export { default as EmailAdminVerication } from './home/EmailAdminVerication';
export {default as VideoSection} from './home/VideoSection';
export {default as Footer} from './Footer';
export {default as Signupmodal} from './home/Signupmodal'
export {default as UploadVideo} from './video/UploadVideo';
export {default as UploadOptions} from './video/UploadOptions';
export {default as OneDrivePicker} from './video/OneDrivePicker';
export {default as Conversion} from './video/Conversion';
export {default as AdvancedOptions} from './video/AdvancedOptions';
export {default as Dashboard} from './dashboard/Dashboard';
export {default as Profile } from './dashboard/Profile';
export {default as RecentTasks} from './dashboard/RecentTasks';
export {default as ChangeEmail } from './dashboard/ChangeEmail';
export {default as ChangePassword } from './dashboard/ChangePassword';
export {default as DeleteAccount } from './dashboard/DeleteAccount';
export {default as DownloadOptions } from './video/DownloadOptions';
export {default as VideoHero } from './videoPage/VideoHero';
export {default as HowToConvert } from './videoPage/HowToConvert';
export {default as FaqItem } from './videoPage/FaqItem';
export {default as FaqSection } from './videoPage/FaqSection';
export { default as Audience } from './home/Audience';
export { default as AudienceCard } from './home/AudienceCard';
export { default as UploadImage } from './image/UploadImage';
export { default as EditImage } from './image/EditImage';
export { default as CropImage } from './image/CropImage';
export {default as ImageThumbnails} from './image/ImageThumbnails';