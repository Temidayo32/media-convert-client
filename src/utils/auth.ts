import { gapi } from 'gapi-script';
import { authClientId, clientId, API_KEY, developerKey, dropboxAppKey  } from '../config/key';
import { DropboxAuth, DropboxResponse } from 'dropbox';
import { Firestore, collection, query, where, getDocs, setDoc, doc, writeBatch, getDoc, deleteDoc, QuerySnapshot, DocumentData } from "firebase/firestore";
import { User, UserCredential, Auth, signInWithCredential, AuthCredential } from "firebase/auth";
import { Task } from '../typings/types';



const SCOPES = 'https://www.googleapis.com/auth/drive.file';
const DROPBOX_CLIENT_ID =  dropboxAppKey;
const STATE = 'random_string';
const DROPBOX_REDIRECT_URI =  `${process.env.REACT_APP_BASE_FRONTEND}/callback`//'http://localhost:3000';

export const validateEmail = (email: string): boolean => {
  const re = /\S+@\S+\.\S+/;
  return re.test(email);
};

export const validatePassword = (password: string): boolean => {
  const passwordRegex = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/;
  return passwordRegex.test(password);
};

export const handleGoogleSignIn = async (): Promise<any> => {
  return new Promise<any>((resolve, reject) => {
    const script = document.createElement('script');
    script.src = 'https://apis.google.com/js/api.js';
    script.onload = () => {
      console.log('Google API script loaded.');
      (window as any).gapi.load('client:auth2', {
        callback: () => {
          (window as any).gapi.auth2.init({
            clientId: authClientId,
          }).then(() => {
            (window as any).gapi.auth2.getAuthInstance().signIn().then((googleUser: any) => {
              resolve(googleUser); // Resolve with googleUser
            }).catch((error: Error) => {
              reject(error);
            });
          }).catch((error: Error) => {
            reject(error);
          });
        },
        onerror: (error: Error) => {
          reject(error);
        }
      });
    };
    script.onerror = (error: Event | string) => {
      console.error('Error loading Google API script:', error);
      reject(error instanceof Error ? error : new Error('Unknown error loading Google API'));
    };
    document.body.appendChild(script);
  });
};


export const initGoogleAPI = async (): Promise<void> => {
  return new Promise<void>((resolve, reject) => {
    const script = document.createElement('script');
    script.src = 'https://apis.google.com/js/api.js';
    script.onload = () => {
      (window as any).gapi.load('client:auth2', {
        callback: () => {
          (window as any).gapi.client.init({
            apiKey: developerKey,
            clientId: clientId,
            discoveryDocs: ['https://www.googleapis.com/discovery/v1/apis/drive/v3/rest'],
            scope: SCOPES
          }).then(() => {
            console.log('Google API initialized');
            resolve();
          }).catch((error: Error) => {
            console.error('Error initializing Google API:', error);
            reject(error);
          });
        },
        onerror: (error: Error) => {
          console.error('Error loading Google API:', error);
          reject(error);
        }
      });
    };
    script.onerror = (error: Event | string) => {
      console.error('Error loading Google API:', error);
      reject(error instanceof Error ? error : new Error('Unknown error loading Google API'));
    };
    document.body.appendChild(script);
  });
};


export const handleGoogleAuth = async (): Promise<string | null> => {
    try {
      console.log('Attempting Google authentication...');
      await initGoogleAPI();
      const authInstance = gapi.auth2.getAuthInstance();
      const authResult = await authInstance.signIn();
      
      // Check if the token has the required scopes
      const currentUser = authInstance.currentUser.get();
      const grantedScopes = currentUser.getGrantedScopes();
      console.log('Granted Scopes:', grantedScopes);
      
      if (!grantedScopes.includes(SCOPES)) {
        console.error('Required scopes not granted.');
        return null;
      }
  
      const token = authResult.xc.access_token;
      console.log('OAuth Token:', token);
      return token;
    } catch (error) {
      console.error('Error during Google authentication:', error);
      return null;
    }
  };
  

  export const handleDropboxAuth = async (): Promise<string> => {
    const dbx = new DropboxAuth({ clientId: DROPBOX_CLIENT_ID });
  
    const authUrl = await dbx.getAuthenticationUrl(
      DROPBOX_REDIRECT_URI!,
      STATE,
      'code',
      'offline',
      undefined,
      'none',
      true
    );
  
    const authWindow = window.open(authUrl.toString(), '_blank', 'width=500,height=700')!;
  
    return new Promise<string>((resolve, reject) => {
      const interval = setInterval(() => {
        try {
          if (authWindow.closed) {
            clearInterval(interval);
            // new Error('Popup closed by user');
          }
  
          const authUrl = new URL(authWindow.location.href);
          if (
            authUrl.origin === window.location.origin &&
            authUrl.searchParams.get('code') &&
            authUrl.searchParams.get('state') === STATE
          ) {
            const code = authUrl.searchParams.get('code');
            authWindow.close();
            clearInterval(interval);
  
            if (code) {
              dbx
              .getAccessTokenFromCode(DROPBOX_REDIRECT_URI!, code)
              .then((response: DropboxResponse<any>) => {
                console.log(response)
                const token = response.result.access_token as string; // Adjust this according to DropboxResponse structure
                resolve(token);
              })
              .catch(reject);
            } else {
              reject(new Error('Authorization failed'));
            }
          }
        } catch (error) {
          // Ignore cross-origin errors until the popup redirects to the redirect URI
        }
      }, 2000);
  
      authWindow.onbeforeunload = () => {
        clearInterval(interval);
        reject(new Error('Popup closed by user'));
      };
    });
  };
  
export async function handleAnonymousUpgradeMergeConflict(
  db: Firestore,
  auth: Auth,
  user: User,
  credential: AuthCredential,
  setIsLoading?: (isLoading: boolean) => void
): Promise<void> {
  const anonymousUserId = user.uid;

  try {
    // Step 1: Copy anonymous user's tasks to a temporary location
    const anonymousTasksRef = query(collection(db, 'tasks'), where('userId', '==', anonymousUserId));
    const temporaryTasksRef = doc(collection(db, 'temporaryTasks'), anonymousUserId);

    // Fetch all tasks for the anonymous user
    const querySnapshot: QuerySnapshot<DocumentData> = await getDocs(anonymousTasksRef);
    const tasks: Task[] = [];
    querySnapshot.forEach((doc) => {
      tasks.push(doc.data() as Task);
    });

    // Store anonymous tasks temporarily
    await setDoc(temporaryTasksRef, { tasks });
    console.log('Step 1');

    // Step 2: Delete the original anonymous user's tasks
    const batchDelete = writeBatch(db);
    querySnapshot.docs.forEach((doc) => {
      batchDelete.delete(doc.ref);
    });
    await batchDelete.commit();
    console.log('Step 2');

    // Step 3: Sign in with the permanent credential
    const userCredential: UserCredential = await signInWithCredential(auth, credential);
    const permanentUserId = userCredential.user.uid;

    // Step 4: Transfer tasks back from temporary location to permanent user
    const docSnapshot = await getDoc(temporaryTasksRef);
    console.log('Step 3');
    const temporaryTasks = docSnapshot.data()?.tasks as Task[];

    // Update userId field for each task
    const batch = writeBatch(db);
    temporaryTasks.forEach((task) => {
      const newTaskRef = doc(collection(db, 'tasks'));
      task.userId = permanentUserId;
      batch.set(newTaskRef, task);
    });

    // Commit the batch operation
    await batch.commit();
    console.log('Step 4');

    // Step 5: Clean up temporary data
    await deleteDoc(temporaryTasksRef);

    console.log('Anonymous user tasks transferred and updated successfully');
  } catch (error) {
    console.error('Error during anonymous upgrade merge conflict handling:', error);
    if(setIsLoading) {
      setIsLoading(false);
    }
  }
}