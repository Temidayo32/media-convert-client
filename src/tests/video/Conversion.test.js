import React from 'react';
import 'regenerator-runtime/runtime';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Conversion } from '../../components';
import { useData } from '../../DataContext';
import { getAuth } from 'firebase/auth';
import { onSnapshot } from 'firebase/firestore';

// Mocking firebase/auth
jest.mock('firebase/auth');

// Mocking firebase/firestore
jest.mock('firebase/firestore', () => {
  const originalModule = jest.requireActual('firebase/firestore');
  return {
    ...originalModule,
    getFirestore: jest.fn(),
    collection: jest.fn(),
    query: jest.fn(),
    where: jest.fn(),
    onSnapshot: jest.fn(),
  };
});

// Mocking socket.io-client
jest.mock('socket.io-client', () => () => ({
  on: jest.fn(),
  off: jest.fn(),
}));

// Mocking DataContext
jest.mock('../../DataContext', () => ({
  useData: jest.fn(),
}));

const mockUseData = {
  uploadedVideos: [
    {
      name: 'test_video.mp4',
      jobId: '1',
      size: '10MB',
      format: 'mp4',
      settings: {},
    },
  ],
  setUploadedVideos: jest.fn(),
};

const mockAuth = {
  currentUser: { uid: '123' },
};

describe('Conversion Component', () => {
  beforeEach(() => {
    useData.mockReturnValue(mockUseData);
    getAuth.mockReturnValue(mockAuth);

    // Mock Firestore onSnapshot to return an unsubscribe function and mock the snapshot
    onSnapshot.mockImplementation((query, callback) => {
      const snapshot = {
        docs: [
          {
            id: '1',
            data: () => ({ progress: 'completed', fileUrl: 'http://example.com/download.mp4' }),
          },
        ],
        docChanges: () => [{ type: 'added', doc: { id: '1', data: () => ({ progress: 'completed', fileUrl: 'http://example.com/download.mp4' }) } }],
      };
      callback(snapshot);
      return jest.fn(); // Unsubscribe function
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders without crashing', () => {
    const { asFragment } = render(<Conversion />);
    expect(screen.getByText(/Converted files are automatically deleted/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('displays uploaded videos', () => {
    const { asFragment } =  render(<Conversion />);
    expect(screen.getByText('test_video.mp4')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('handles remove video action', () => {
    const { asFragment } =  render(<Conversion />);
    fireEvent.click(screen.getByTitle('Delete'));
    expect(mockUseData.setUploadedVideos).toHaveBeenCalled();
    expect(asFragment()).toMatchSnapshot();
  });

  test('shows completed progress for uploaded videos', () => {
    useData.mockReturnValue({
      ...mockUseData,
      uploadedVideos: [
        {
          ...mockUseData.uploadedVideos[0],
          progress: 'completed',
          fileUrl: 'http://example.com/download.mp4',
        },
      ],
    });

    const { asFragment } =  render(<Conversion />);
    expect(screen.getByText('completed')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
