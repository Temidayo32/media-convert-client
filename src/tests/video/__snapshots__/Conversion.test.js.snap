// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conversion Component displays uploaded videos 1`] = `
<DocumentFragment>
  <div
    class="container mx-0 lg:mx-auto w-full lg:w-4/5 bg-white p-4 sm:p-8 rounded-lg shadow-lg"
  >
    <ul
      class="space-y-4"
    >
      <li
        class="flex flex-col gap-4 sm:flex-row sm:gap-0 justify-between items-center border border-gray-300 p-4 rounded-lg"
      >
        <div
          class="flex flex-col sm:flex-row w-full md:w-8/12 justify-center"
        >
          <div
            class="flex flex-col w-full sm:w-7/12"
          >
            <span
              class="text-teal-800 text-center sm:text-left mb-4 sm:mb-0 text-sm lg:text-base font-semibold"
            >
              test_video.mp4
            </span>
          </div>
          <div
            class="flex items-center sm:ml-12 lg:ml-0 lg:justify-start justify-center w-full sm:w-5/12 mt-2 sm:mt-0"
          >
            <div
              class="relative w-3/5 sm:w-48 h-4 bg-gray-200 rounded-full overflow-hidden"
            >
              <div
                class="absolute left-0 top-0 bg-green-500 h-full rounded-full"
                style="width: 100%; transition: width 4s ease-in-out; animation: none;"
              />
              <span
                class="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 text-white text-xs sm:text-sm"
              >
                completed
              </span>
            </div>
          </div>
        </div>
        <div
          class="flex items-center justify-center sm:justify-end sm:ml-2 gap-6 w-full md:w-4/12 mt-4 sm:mt-0"
        >
          <div
            class="relative inline-block text-left"
          >
            <div
              aria-expanded="true"
              aria-haspopup="true"
              class="py-2 sm:px-4 px-8 md:px-8 lg:px-4 flex items-center justify-center lg:justify-start text-center gap-4 lg:gap-6 w-full rounded-t-lg transition-colors duration-300 bg-teal-500 hover:bg-teal-600 shadow-2xl drop-shadow-2xl text-white"
              id="options-menu"
            >
              <button
                class="text-sm sm:text-lg lg:text-xl xl:text-2xl text-white"
                type="button"
              >
                Download
              </button>
              <div
                class="hidden lg:block lg:h-10 p-0 lg:w-0.5 bg-white"
              />
              <svg
                class="hidden lg:block lg:text-2xl text-white cursor-pointer"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm289.1-43.4c7.5-7.5 19.8-7.5 27.3 0 3.8 3.8 5.6 8.7 5.6 13.6s-1.9 9.9-5.7 13.7l-94.3 94c-7.6 6.9-19.3 6.7-26.6-.6l-95.7-95.4c-7.5-7.5-7.6-19.7 0-27.3 7.5-7.5 19.7-7.6 27.3 0l81.1 81.9 81-79.9z"
                />
              </svg>
            </div>
          </div>
          <button
            title="Delete"
          >
            <svg
              class="text-2xl text-gray-500 hover:text-red-600 transform hover:scale-110 transition-colors duration-300"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </div>
      </li>
    </ul>
    <p
      class="text-gray-400 text-xs md:text-sm lg:text-base mt-4 text-center sm:text-left"
    >
      Converted files are automatically deleted after 8 hours to protect your privacy. Please download your files before they are deleted.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Conversion Component handles remove video action 1`] = `
<DocumentFragment>
  <div
    class="container mx-0 lg:mx-auto w-full lg:w-4/5 bg-white p-4 sm:p-8 rounded-lg shadow-lg"
  >
    <ul
      class="space-y-4"
    >
      <li
        class="flex flex-col gap-4 sm:flex-row sm:gap-0 justify-between items-center border border-gray-300 p-4 rounded-lg"
      >
        <div
          class="flex flex-col sm:flex-row w-full md:w-8/12 justify-center"
        >
          <div
            class="flex flex-col w-full sm:w-7/12"
          >
            <span
              class="text-teal-800 text-center sm:text-left mb-4 sm:mb-0 text-sm lg:text-base font-semibold"
            >
              test_video.mp4
            </span>
          </div>
          <div
            class="flex items-center sm:ml-12 lg:ml-0 lg:justify-start justify-center w-full sm:w-5/12 mt-2 sm:mt-0"
          >
            <div
              class="relative w-3/5 sm:w-48 h-4 bg-gray-200 rounded-full overflow-hidden"
            >
              <div
                class="absolute left-0 top-0 bg-green-500 h-full rounded-full"
                style="width: 100%; transition: width 4s ease-in-out; animation: none;"
              />
              <span
                class="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 text-white text-xs sm:text-sm"
              >
                completed
              </span>
            </div>
          </div>
        </div>
        <div
          class="flex items-center justify-center sm:justify-end sm:ml-2 gap-6 w-full md:w-4/12 mt-4 sm:mt-0"
        >
          <div
            class="relative inline-block text-left"
          >
            <div
              aria-expanded="true"
              aria-haspopup="true"
              class="py-2 sm:px-4 px-8 md:px-8 lg:px-4 flex items-center justify-center lg:justify-start text-center gap-4 lg:gap-6 w-full rounded-t-lg transition-colors duration-300 bg-teal-500 hover:bg-teal-600 shadow-2xl drop-shadow-2xl text-white"
              id="options-menu"
            >
              <button
                class="text-sm sm:text-lg lg:text-xl xl:text-2xl text-white"
                type="button"
              >
                Download
              </button>
              <div
                class="hidden lg:block lg:h-10 p-0 lg:w-0.5 bg-white"
              />
              <svg
                class="hidden lg:block lg:text-2xl text-white cursor-pointer"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm289.1-43.4c7.5-7.5 19.8-7.5 27.3 0 3.8 3.8 5.6 8.7 5.6 13.6s-1.9 9.9-5.7 13.7l-94.3 94c-7.6 6.9-19.3 6.7-26.6-.6l-95.7-95.4c-7.5-7.5-7.6-19.7 0-27.3 7.5-7.5 19.7-7.6 27.3 0l81.1 81.9 81-79.9z"
                />
              </svg>
            </div>
          </div>
          <button
            title="Delete"
          >
            <svg
              class="text-2xl text-gray-500 hover:text-red-600 transform hover:scale-110 transition-colors duration-300"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </div>
      </li>
    </ul>
    <p
      class="text-gray-400 text-xs md:text-sm lg:text-base mt-4 text-center sm:text-left"
    >
      Converted files are automatically deleted after 8 hours to protect your privacy. Please download your files before they are deleted.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Conversion Component renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="container mx-0 lg:mx-auto w-full lg:w-4/5 bg-white p-4 sm:p-8 rounded-lg shadow-lg"
  >
    <ul
      class="space-y-4"
    >
      <li
        class="flex flex-col gap-4 sm:flex-row sm:gap-0 justify-between items-center border border-gray-300 p-4 rounded-lg"
      >
        <div
          class="flex flex-col sm:flex-row w-full md:w-8/12 justify-center"
        >
          <div
            class="flex flex-col w-full sm:w-7/12"
          >
            <span
              class="text-teal-800 text-center sm:text-left mb-4 sm:mb-0 text-sm lg:text-base font-semibold"
            >
              test_video.mp4
            </span>
          </div>
          <div
            class="flex items-center sm:ml-12 lg:ml-0 lg:justify-start justify-center w-full sm:w-5/12 mt-2 sm:mt-0"
          >
            <div
              class="relative w-3/5 sm:w-48 h-4 bg-gray-200 rounded-full overflow-hidden"
            >
              <div
                class="absolute left-0 top-0 bg-green-500 h-full rounded-full"
                style="width: 100%; transition: width 4s ease-in-out; animation: none;"
              />
              <span
                class="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 text-white text-xs sm:text-sm"
              >
                completed
              </span>
            </div>
          </div>
        </div>
        <div
          class="flex items-center justify-center sm:justify-end sm:ml-2 gap-6 w-full md:w-4/12 mt-4 sm:mt-0"
        >
          <div
            class="relative inline-block text-left"
          >
            <div
              aria-expanded="true"
              aria-haspopup="true"
              class="py-2 sm:px-4 px-8 md:px-8 lg:px-4 flex items-center justify-center lg:justify-start text-center gap-4 lg:gap-6 w-full rounded-t-lg transition-colors duration-300 bg-teal-500 hover:bg-teal-600 shadow-2xl drop-shadow-2xl text-white"
              id="options-menu"
            >
              <button
                class="text-sm sm:text-lg lg:text-xl xl:text-2xl text-white"
                type="button"
              >
                Download
              </button>
              <div
                class="hidden lg:block lg:h-10 p-0 lg:w-0.5 bg-white"
              />
              <svg
                class="hidden lg:block lg:text-2xl text-white cursor-pointer"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm289.1-43.4c7.5-7.5 19.8-7.5 27.3 0 3.8 3.8 5.6 8.7 5.6 13.6s-1.9 9.9-5.7 13.7l-94.3 94c-7.6 6.9-19.3 6.7-26.6-.6l-95.7-95.4c-7.5-7.5-7.6-19.7 0-27.3 7.5-7.5 19.7-7.6 27.3 0l81.1 81.9 81-79.9z"
                />
              </svg>
            </div>
          </div>
          <button
            title="Delete"
          >
            <svg
              class="text-2xl text-gray-500 hover:text-red-600 transform hover:scale-110 transition-colors duration-300"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </div>
      </li>
    </ul>
    <p
      class="text-gray-400 text-xs md:text-sm lg:text-base mt-4 text-center sm:text-left"
    >
      Converted files are automatically deleted after 8 hours to protect your privacy. Please download your files before they are deleted.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Conversion Component shows completed progress for uploaded videos 1`] = `
<DocumentFragment>
  <div
    class="container mx-0 lg:mx-auto w-full lg:w-4/5 bg-white p-4 sm:p-8 rounded-lg shadow-lg"
  >
    <ul
      class="space-y-4"
    >
      <li
        class="flex flex-col gap-4 sm:flex-row sm:gap-0 justify-between items-center border border-gray-300 p-4 rounded-lg"
      >
        <div
          class="flex flex-col sm:flex-row w-full md:w-8/12 justify-center"
        >
          <div
            class="flex flex-col w-full sm:w-7/12"
          >
            <span
              class="text-teal-800 text-center sm:text-left mb-4 sm:mb-0 text-sm lg:text-base font-semibold"
            >
              test_video.mp4
            </span>
          </div>
          <div
            class="flex items-center sm:ml-12 lg:ml-0 lg:justify-start justify-center w-full sm:w-5/12 mt-2 sm:mt-0"
          >
            <div
              class="relative w-3/5 sm:w-48 h-4 bg-gray-200 rounded-full overflow-hidden"
            >
              <div
                class="absolute left-0 top-0 bg-green-500 h-full rounded-full"
                style="width: 100%; transition: width 4s ease-in-out; animation: none;"
              />
              <span
                class="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 text-white text-xs sm:text-sm"
              >
                completed
              </span>
            </div>
          </div>
        </div>
        <div
          class="flex items-center justify-center sm:justify-end sm:ml-2 gap-6 w-full md:w-4/12 mt-4 sm:mt-0"
        >
          <div
            class="relative inline-block text-left"
          >
            <div
              aria-expanded="true"
              aria-haspopup="true"
              class="py-2 sm:px-4 px-8 md:px-8 lg:px-4 flex items-center justify-center lg:justify-start text-center gap-4 lg:gap-6 w-full rounded-t-lg transition-colors duration-300 bg-teal-500 hover:bg-teal-600 shadow-2xl drop-shadow-2xl text-white"
              id="options-menu"
            >
              <button
                class="text-sm sm:text-lg lg:text-xl xl:text-2xl text-white"
                type="button"
              >
                Download
              </button>
              <div
                class="hidden lg:block lg:h-10 p-0 lg:w-0.5 bg-white"
              />
              <svg
                class="hidden lg:block lg:text-2xl text-white cursor-pointer"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm289.1-43.4c7.5-7.5 19.8-7.5 27.3 0 3.8 3.8 5.6 8.7 5.6 13.6s-1.9 9.9-5.7 13.7l-94.3 94c-7.6 6.9-19.3 6.7-26.6-.6l-95.7-95.4c-7.5-7.5-7.6-19.7 0-27.3 7.5-7.5 19.7-7.6 27.3 0l81.1 81.9 81-79.9z"
                />
              </svg>
            </div>
          </div>
          <button
            title="Delete"
          >
            <svg
              class="text-2xl text-gray-500 hover:text-red-600 transform hover:scale-110 transition-colors duration-300"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </button>
        </div>
      </li>
    </ul>
    <p
      class="text-gray-400 text-xs md:text-sm lg:text-base mt-4 text-center sm:text-left"
    >
      Converted files are automatically deleted after 8 hours to protect your privacy. Please download your files before they are deleted.
    </p>
  </div>
</DocumentFragment>
`;
